tools/clang/lib/Sema/AnalysisBasedWarnings.cpp

```diff
@@ -2006,9 +2017,13 @@ AnalysisBasedWarnings::IssueWarnings(sema::AnalysisBasedWarnings::Policy P,
     Analyzer.run(AC);
   }
 
+  // HLSL Change Start - Generate warnings for uninitialized out params.
   if (!Diags.isIgnored(diag::warn_uninit_var, D->getLocStart()) ||
       !Diags.isIgnored(diag::warn_sometimes_uninit_var, D->getLocStart()) ||
-      !Diags.isIgnored(diag::warn_maybe_uninit_var, D->getLocStart())) {
+      !Diags.isIgnored(diag::warn_maybe_uninit_var, D->getLocStart()) ||
+      !Diags.isIgnored(diag::warn_hlsl_uninit_out_param, D->getLocStart()) ||
+      !Diags.isIgnored(diag::warn_hlsl_sometimes_uninit_out_param, D->getLocStart())) {
+    // HLSL Change End  - Generate warnings for uninitialized out params.
```

Comment:
This looks like if we enable one of these, we get them all. I'm not saying that's wrong, but I want to be sure we're consciously making that decision

Reply:
If you enable one of those warnings you need to perform the analysis (which also generates them). Calling `Diag(...)` on an ignored diagnostic still suppresses the message, so you'll only get the output from the ones that are enabled.

This check exists to skip the AST traversal analysis if all the warnings generated by it are ignored.

Reply:
TIL that! Very useful information for something else I'm working on!